### PR TITLE
Add new recipe Inverted Repeats Finder (IRF)

### PR DESCRIPTION
**Pull request by wolf14353250**
Add new recipe Inverted Repeats Finder (IRF):

An inverted repeat (IR) consists of two arms (or stems) of similar DNA or RNA—with one inverted and complemented relative to the other—adjacent to a central, usually nonhomologous, spacer sequence.

This program detects approximate inverted repeats. It can be tuned, using the parameters, to target different stem lengths, different spacer lengths, and different degrees of approximation. It allows GT pairing, which occurs in RNA stem-loops. It also allows detection of mirror repeats (where the stems are inverted, but not complemented). This is useful for establishing background occurrence for statistical analysis of IRs.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by wolf14353250** [comment]
@BiocondaBot please add label

File: recipes/irf/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+mkdir -p "${PREFIX}"/bin
+cp bin/irf308.linux.exe "${PREFIX}"/bin/irf
```
Comment (bgruening) [inline recipes/irf/build.sh:3]: This is not recommended.. can we help you to compile the package with our tool chain?

Comment (wolf14353250) [reply recipes/irf/build.sh:3]: Hi, I tried to use the instructions of compiling in https://github.com/Benson-Genomics-Lab/IRF in build.sh.
`cd src
make`
and specified {{ compiler('c') }} in the meta.yaml.

But the test pipeline raised error:
`gcc -c -o irf.o  irf.3.c -O2
make: gcc: No such file or directory
make: *** [Makefile:44: irf.o] Error 127`

I am grateful if you can give us some suggestions to help us compile the package, thank you!

Comment (bgruening) [reply recipes/irf/build.sh:3]: You should not hardcode gcc in your makefile. On Linux we might use gcc but on OSX clang. Good practice is to use $CC in your makefile as an env variable. Then the host system can set compliers as they like. You can default to gcc but make sure an env variable can overwrite your default complier

Comment (wolf14353250) [reply recipes/irf/build.sh]: Thanks for your help. I modified build.sh accordingly and passed the test pipeline. Please view.

**Comment by bgruening** [comment]
It might work now also for OSX. Can you try that?

**Review by bgruening — APPROVED** [review]

**Comment by wolf14353250** [comment]
@bgruening Hi, please merge, thank you

**Comment by bgruening** [comment]
Thanks @wolf14353250

**Pull request merged**